pkg/cri/server/container_create_linux.go

```diff
@@ -219,12 +219,11 @@ func (c *criService) containerSpec(
 		}
 		specOpts = append(specOpts, oci.WithPrivileged)
 		if !ociRuntime.PrivilegedWithoutHostDevices {
-			specOpts = append(specOpts, oci.WithHostDevices, oci.WithAllDevicesAllowed)
-		} else {
-			// add requested devices by the config as host devices are not automatically added
-			specOpts = append(specOpts, customopts.WithDevices(c.os, config, c.config.DeviceOwnershipFromSecurityContext),
-				customopts.WithCapabilities(securityContext, c.allCaps))
+			specOpts = append(specOpts, oci.WithHostDevices)
 		}
+		specOpts = append(specOpts, customopts.WithDevices(c.os, config, c.config.DeviceOwnershipFromSecurityContext),
+			customopts.WithCapabilities(securityContext, c.allCaps))
+		specOpts = append(specOpts, oci.WithAllDevicesAllowed)
```

Comment:
should WithAllDevicesAllowed be set when PrivilegedWithoutHostDevices is true?

Reply:
fixed: add devices first, only PrivilegedWithoutHostDevices == false will add `withHostDevices` and `WithAllDevicesAllowed`